### PR TITLE
Commit and job info debug dump

### PR DESCRIPTION
**Pull request by brycemcanally**
Adds commit and job info collection to debug dump and an additional flag (`--limit`) to limit the number of commits / jobs returned. ~~Also, it looks like there was a change to our proto code generation that got merged without committing the generated go files, so that is why there are a lot of `.pb.go` files in this PR.~~

### CODE REVIEW

File: src/server/debug/server/server.go [hunk 1]
```diff
@@ -347,6 +355,9 @@ func redirectBinaryFunc(ctx context.Context) redirectFunc {
 }
 
 func (s *debugServer) Dump(request *debug.DumpRequest, server debug.Debug_DumpServer) error {
+	if request.Limit == 0 {
```
Comment (Tryneus) [inline src/server/debug/server/server.go:358]: drive-by comment: I feel like `0` should be zero commits and `-1` should be unlimited, otherwise there is no way to specify `0`. but I suppose golang default values make this difficult since we want to default behavior to be unlimited

File: src/client/debug/debug.proto [hunk 2]
```diff
@@ -37,6 +37,7 @@ message BinaryRequest {
 
 message DumpRequest {
   Filter filter = 1;
+  int64 limit = 2;
```
Comment (jdoliner) [inline src/client/debug/debug.proto:40]: This limits the number of commits and jobs returned right? I think maybe a slightly more specific name than limit would be good. Just calling it `limit` makes it seem like it's a global limit on the number of bytes returned.

**Comment by jdoliner** [comment]
I think you might have an older version of the proto image. The changes to the proto file look like the opposite of what I just got when I updated.

**Comment by brycemcanally** [comment]
Yeah, I ran it with `--no-cache` and got what is in master.

File: src/server/debug/server/server.go (hunk 1 above)
Comment (brycemcanally) [reply src/server/debug/server/server.go:358]: Yeah, I don't really have a strong opinion either way. I was aiming to be consistent with list commit which interprets 0 to mean all of them and it is nice to be able to use the Go default behavior, but like you said we cannot specify 0 with this. What do you think @jdoliner ?

File: src/client/debug/debug.proto (hunk 2 above)
Comment (brycemcanally) [reply src/client/debug/debug.proto:40]: Yeah, I wasn't really sure what to call this considering it applies to commit and job counts. I actually don't have a good name in mind because `commit_and_job_limit` seems more confusing and splitting it up into two fields seems like it would decrease the ease of use because it seems unlikely that a user will want to have limits that are specific to the commits and jobs. I think `limit` might be fine if I just add a note in the proto (the cmd already has a description that I think is adequate).

Comment (jdoliner) [reply src/client/debug/debug.proto:40]: Yeah a note in the protos should be fine. I don't have a better name either.

File: src/server/debug/server/server.go (hunk 1 above)
Comment (jdoliner) [reply src/server/debug/server/server.go:358]: I think having 0 mean all of them is probably easiest. I don't think there's much need to be able to do a dump that doesn't get any commits or jobs. The need seems small enough that we should just do the easy thing.

**Review by jdoliner — APPROVED** [review]
LGTM
